### PR TITLE
Documentation refactoring to align to devonfw pattern.

### PR DESCRIPTION
**Pull request by isandesh1986**
- Fixed broken images/resources include directive issue. 
- broke down single master solicitor to multiple sub asciidoc files.
- added pom.xml in documentation folder to align with devon
THIS PR NEEDS A CHANGE IN GITHUB ACTIONS - RELEASE. DO NOT MERGE UNTIL RELEASE YML IS UPDATED ACCORDINGLY. 
In order to align with devonfw standards - had to rename the PDF file as master-solicitor.pdf. Same needs to be
updated in RELEASE.YML for GITHUB ACTIONS.

### CODE REVIEW

**Review by ohecker — CHANGES_REQUESTED** [review]
The current refactoring again breaks the Solicitor feature of having the userguide included in the executable jar. So this would again cause the same problem as documented in https://github.com/devonfw/solicitor/issues/70 which happened on the first attempt to include the Solicitor documentation into devon. 
Having the user guide packaged within the JAR is a major feature to ensure that users which have the executable also have access to the correct version of the userguide.

For reference how it should look like:

- download the latest release of solicitor from https://github.com/devonfw/solicitor/releases/download/v1.2.0/solicitor.jar
- run it with _java -jar solicitor.ha -eug_ (eug = Extract user guide)
- The userguide (and the license info) is extracted into the current directory
- Review the userguide: This is the reference and this should still look the same after documentation refactoring. Also observe the Version Info (including build date and commit hash) which is automatically included in the document within the build process.

Note: I did not review the documentation refactoring (modularization) in more detail up to now. I will do so at a later stage once the build issue is solved. (But my expectation is: refactoring does not alter how the userguide looks like for the end user.)

**Comment by hohwille** [comment]
Hi there,
can you get some inspiration from devonfw-ide, please?
It is solving all the issues you are discussion here already.
https://github.com/devonfw/ide/
https://github.com/devonfw/ide/blob/master/scripts/pom.xml#L103

File: pom.xml [hunk 1]
```diff
@@ -211,76 +211,6 @@
 					<encoding>utf-8</encoding>
 				</configuration>
 			</plugin>
-			<plugin>
```
[the remark this answers is not in the document]
Comment (isandesh1986) [reply pom.xml:214]: Agreed. Changes are done accordingly. Please review.

**Comment by isandesh1986** [comment]
> Hi there,
> can you get some inspiration from devonfw-ide, please?
> It is solving all the issues you are discussion here already.
> https://github.com/devonfw/ide/
> https://github.com/devonfw/ide/blob/master/scripts/pom.xml#L103

Thanks Jorg. I was already referring devon4j also. Achieved the same in a slightly different way.

**Comment by ohecker** [comment]
Hi @isandesh1986. I had a look onto your changes.

I could not build it with "mvn clean install" (this should in any case work and generate the executable JAR in the target directory:

- pom of documentation references "devonfw-docgen-all" in  a SNAPSHOT version unvailable from maven central
- pom of core references solicitor-starter in version 1.2.0-SNAPSHOT instead of 1.3.0-SNAPSHOT

Even after fixing this the build failed in test SolicitorCreateOwnLicenseInfoTest. Within this test Solicitor is used to process its own license info to create the license list which needs to be included in the release.

To what I have seen the test fails because some input file is not existing because the maven plugin which creates it is skipped:

```
 --- license-maven-plugin:1.14:aggregate-download-licenses (default) @ solicitor ---
[INFO] skip flag is on, will skip goal. 
```
This "skip" seems to be a side effect of making the whole project a multimodule project.

Other things I noticed:
* You commited some file in directory "output". This is some data created probably by tests and should not be committed to Git. Same holds for file solicitor.log.2021-02-03.0.gz in the root directory
* The POM configuration which is used to correctly set the version info in the documentation is contained in the POM of core. Probably it will not have any effect there.
* That header boilerplate which shows the current version number / build data in the PDF seems to be missing in the docs anyway
* Main headings are missing in the generated PDF
* There are a lot of ERRORS and WARNINGS when processing the Asciidoc

File: core/pom.xml [hunk 2]
```diff
@@ -0,0 +1,300 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
```
Comment (hohwille) [inline core/pom.xml:4]: devonfw indendation is 2 spaces and not 4 ;)

File: core/src/main/java/com/devonfw/tools/solicitor/common/ResourceToFileCopier.java [hunk 3]
```diff
@@ -205,10 +205,10 @@ public String copyReourcesToFile(ResourceGroup resourceGroup, String targetDir)
         switch (resourceGroup) {
         case USERGUIDE:
             new CopySequenceBuilder()
-                    .withCopyOperation("classpath:solicitor_userguide.pdf", "target/solicitor_userguide.pdf")
+                    .withCopyOperation("classpath:master-solicitor.pdf", "target/master-solicitor.pdf")
                     .withCopyOperation("classpath:solicitor_licenseinfo.html", "target/solicitor_licenseinfo.html")
                     .replaceInTarget("target", targetDir).execute();
-            returnString = "solicitor_userguide.pdf";
+            returnString = "master-solicitor.pdf";
```
Comment (hohwille) [inline core/src/main/java/com/devonfw/tools/solicitor/common/ResourceToFileCopier.java:211]: I think the filename does not have to be changed. IMHO `solicitor_userguide` makes much more sense than `master-solicitor`.

Comment (isandesh1986) [reply core/src/main/java/com/devonfw/tools/solicitor/common/ResourceToFileCopier.java:211]: Agreed. just to align with rest of the devon document repos - had to make it master-solicitor.

Comment (maybeec) [reply core/src/main/java/com/devonfw/tools/solicitor/common/ResourceToFileCopier.java]: I think there is actually no reason for that. None of our algorithms rely on that behavior to have this specific name.

**Comment by ohecker** [comment]
So this PR is stale (open and unchanged for more than a year). Actually I believe the documentation integration with devonfw is still broken. I will close this now. Please feel free to resubmit it once it neither breaks embedding of the userguide into the executable jar nor otherwise  affects the generated document.

**Pull request closed without merging**